src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -695,8 +700,12 @@ impl<'cfg> RustcTargetData<'cfg> {
         if requested_kinds.iter().any(CompileKind::is_host) {
             let ct = CompileTarget::new(&rustc.host)?;
             target_info.insert(ct, host_info.clone());
-            target_config.insert(ct, host_config.clone());
-        }
+            if target_applies_to_host {
```

Comment:
Hm so I still have thoughts on this, but my previous comments keep getting minimized and now I can't unminimize them.

I see what this is doing now, but I think it should probably just always call `target_cfg_triple` here instead of trying to conditionally reuse the `host_config`. I think that's the intention here, right?

Reply:
>  I think it should probably just always call `target_cfg_triple` here instead of trying to conditionally reuse the host_config

I had considered that, however I didn't use that approach since it wasn't 100% clear if `target_cfg_triple` was fully deterministic to the point where under all conditions calling it multiple times would always provide the same results, so I cloned it to make that easier to reason about.

Reply:
I believe it's deterministic, so it's fine to call that here.